Xamarin.Forms.Core/BindingExpression.cs

```diff
@@ -371,6 +372,16 @@ bool TryConvert(BindingExpressionPart part, ref object value, Type convertTo, bo
 			object original = value;
 			try
 			{
+				var stringValue = value as string ?? string.Empty;
+				// see: https://bugzilla.xamarin.com/show_bug.cgi?id=32871
+				// do not canonicalize "*.[.]"; "1." should not update bound BindableProperty
+				if (stringValue.EndsWith(".") && DecimalTypes.Contains(convertTo))
```

Comment:
Could do that. I don't have a great reason for one over the other.. Precedent? The current flow throws an exceptions so I stuck with that flow. In general, if I were starting from scratch, I'd avoid flows that throw (e.g. search out for a Convert.TryChangeType or equivalent) but when fixing bugs, I try to adopt whatever convention exists.